### PR TITLE
Hide generated frames with hidden annotation

### PR DESCRIPTION
**Pull request by fengxue-IS**
- map LambdaForm$Hidden / vm.annotation.Hidden to HIDDEN_ANNOTATION
- Reuse J9AccMethodFrameIteratorSkip flag for HiddenAnnotation
- ifdef JIT checks on J9AccMethodFrameIteratorSkip

Replaces: #12672 
Fixes: #11872, Fixes: #7741

Signed-off-by: Jack Lu <Jack.S.Lu@ibm.com>

### CODE REVIEW

**Comment by fengxue-IS** [comment]
@babsingh can you please take a look

File: runtime/vm/swalk.c [hunk 1]
```diff
@@ -394,6 +394,19 @@ UDATA  walkStackFrames(J9VMThread *currentThread, J9StackWalkState *walkState)
 
 UDATA walkFrame(J9StackWalkState * walkState)
 {
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+	{
```
Comment (babsingh) [inline runtime/vm/swalk.c:398]: Is the extra scope required?

**Review by babsingh — CHANGES_REQUESTED** [review]
Can you confirm if ...
- no regression is seen in personal builds?
- #11872 and #7741 are fixed?

File: runtime/vm/swalk.c [hunk 2]
```diff
@@ -394,6 +394,19 @@ UDATA  walkStackFrames(J9VMThread *currentThread, J9StackWalkState *walkState)
...
+		J9ROMMethod * romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(walkState->method);
+		/* Check for extended modifiers on the ROM method */
+		if ((NULL != romMethod) && J9ROMMETHOD_HAS_EXTENDED_MODIFIERS(romMethod)) {
+			UDATA extraModifiers = getExtendedModifiersDataFromROMMethod(romMethod);
```
[the remark this answers is not in the document]
Comment (fengxue-IS) [reply runtime/vm/swalk.c]: fixed

File: runtime/bcutil/ClassFileOracle.cpp [hunk 3]
```diff
@@ -61,6 +61,15 @@ ClassFileOracle::KnownAnnotation ClassFileOracle::_knownAnnotations[] = {
 #define VALUEBASED_SIGNATURE "Ljdk/internal/ValueBased;"
 		{VALUEBASED_SIGNATURE, sizeof(VALUEBASED_SIGNATURE)},
 #undef VALUEBASED_SIGNATURE
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+#if JAVA_SPEC_VERSION >= 16
+#define HIDDEN_SIGNATURE "Ljdk/internal/vm/annotation/Hidden;"
+#else /* JAVA_SPEC_VERSION >= 16 */
+#define HIDDEN_SIGNATURE "Ljava/lang/invoke/LambdaForm$Hidden;"
+#endif /* JAVA_SPEC_VERSION >= 16 */
+	{HIDDEN_SIGNATURE, sizeof(HIDDEN_SIGNATURE)},
```
[the remark this answers is not in the document]
Comment (fengxue-IS) [reply runtime/bcutil/ClassFileOracle.cpp]: fixed

File: runtime/vm/swalk.c [hunk 4]
```diff
@@ -394,6 +394,19 @@ UDATA  walkStackFrames(J9VMThread *currentThread, J9StackWalkState *walkState)
 
 UDATA walkFrame(J9StackWalkState * walkState)
 {
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+	{
+		J9ROMMethod * romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(walkState->method);
```
[the remark this answers is not in the document]
Comment (fengxue-IS) [reply runtime/vm/swalk.c]: fixed

**Comment by tajila** [comment]
Is this ready for another review @fengxue-IS ?

**Comment by fengxue-IS** [comment]
> Is this ready for another review @fengxue-IS ?

Yes, can you take a look please

**Comment by tajila** [comment]
@gacholio Please review these changes

File: debugtools/DDR_VM/data/superset-constants.dat [hunk 5]
```diff
@@ -1130,6 +1131,7 @@ C|CFR_METHOD_ACCESS_NEWJDK5_MASK
 C|CFR_METHOD_EXT_HAS_CODE_TYPE_ANNOTATIONS
 C|CFR_METHOD_EXT_HAS_METHOD_TYPE_ANNOTATIONS
 C|CFR_METHOD_EXT_INVALID_CP_ENTRY
+C|CFR_METHOD_EXT_HIDDEN_ANNOTATIONS
```
Comment (gacholio) [inline debugtools/DDR_VM/data/superset-constants.dat:1134]: Is this intentionally changed?

Comment (gacholio) [reply debugtools/DDR_VM/data/superset-constants.dat:1134]: I thought this file was generated.

**Comment by gacholio** [comment]
Copyright may need to be updated - the link goes nowhere so it's hard to say.

**Comment by gacholio** [comment]
This is not the correct way to do this - the stack walker should not be skipping these frames, it should be done inside the frame iterator. There are many cases where we want the true view of the stack.

**Comment by fengxue-IS** [comment]
> This is not the correct way to do this - the stack walker should not be skipping these frames, it should be done inside the frame iterator. There are many cases where we want the true view of the stack.

@Gac can you give a bit more insight as to which type of iterators should skip these frames (beyond the getCallerClass interators) or which shouldn't?

**Comment by gacholio** [comment]
Anyone dealing with hidden frames of any kind needs to be updated. Now would be a good time to commonize the code where possible.

Suggest searching for `J9AccMethodFrameIteratorSkip` and `jlrMethodInvoke`.

**Comment by gacholio** [comment]
Is this not identical to `FrameIteratorSkip`? How does this differ?

**Comment by fengxue-IS** [comment]
> Is this not identical to `FrameIteratorSkip`? How does this differ?
I am not sure if FrameIteratorSkip and Hidden are the same thing as JIT seems to be using `J9AccMethodFrameIteratorSkip` and ROMClassWriter as well. So I wasn't sure if these can be just mapped to FrameIteratorSkip

**Comment by gacholio** [comment]
As long as we can recreate the class file from the ROM class with the correct annotations, I can't see any immediate reason we can't re-use the skip tag bit.

**Comment by fengxue-IS** [comment]
I have updated the code as Gac suggested to reuse the `J9AccMethodFrameIteratorSkip` flag which OpenJ9 MH used for skipping internal methods. ifdefed out the JIT code that checks this flag under OpenJDK MH as it nolonger identify `thunk_archetype` methods

@tajila can you take a look at this

**Review by tajila — APPROVED** [review]

**Comment by tajila** [comment]
@0xdaryl Please review the JIT changes

**Review by 0xdaryl — APPROVED** [review]

**Review by gacholio — APPROVED** [review]

**Comment by gacholio** [comment]
jenkins test sanity zlinux jdk8

**Comment by gacholio** [comment]
jenkins test sanity winojdk292,xlinuxojdk292 jdk16 depends ibmruntimes/openj9-openjdk-jdk16#29

**Comment by gacholio** [comment]
jenkins test sanity,extended plinux jdk11

**Pull request merged**